### ISSUE TITLE
Error using Jaxb to generate WS Client on 2.0.0.RELEASE

### ISSUE THREAD

**Issue by lasalazarr** [spring-projects/spring-boot#12301]
After update from Spring Boot 2.0.0.RC1 to 2.0.0.RELEASE, doesn't work a simple project to Consuming a SOAP web service. 

To create the project you can follow the blog: 
https://spring.io/guides/gs/consuming-web-service/

Error
```
* What went wrong:
Execution failed for task ':customer-integration:genJaxb'.
> java.lang.Error: java.lang.reflect.InvocationTargetException
```

The stacktrace is:

````
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at com.sun.tools.xjc.reader.Ring.get(Ring.java:102)
        ... 70 more
Caused by: java.lang.Error: java.lang.reflect.InvocationTargetException
        at com.sun.tools.xjc.reader.Ring.get(Ring.java:113)
        at com.sun.tools.xjc.reader.xmlschema.BindingComponent.getClassSelector(BindingComponent.java:61)
        at com.sun.tools.xjc.reader.xmlschema.ColorBinder.<init>(ColorBinder.java:58)
        at com.sun.tools.xjc.reader.xmlschema.BindGreen.<init>(BindGreen.java:59)
        ... 74 more
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at com.sun.tools.xjc.reader.Ring.get(Ring.java:102)
        ... 77 more
Caused by: java.lang.NoClassDefFoundError: javax/activation/MimeTypeParseException
        at com.sun.tools.xjc.reader.Ring.get(Ring.java:100)
        at com.sun.tools.xjc.reader.xmlschema.DefaultClassBinder.<init>(DefaultClassBinder.java:92)
        at com.sun.tools.xjc.reader.xmlschema.ClassSelector.<init>(ClassSelector.java:209)
        ... 81 more
Caused by: java.lang.ClassNotFoundException: javax.activation.MimeTypeParseException
        at org.apache.tools.ant.AntClassLoader.findClassInComponents(AntClassLoader.java:1386)
        at org.apache.tools.ant.AntClassLoader.findClass(AntClassLoader.java:1335)
        at org.apache.tools.ant.AntClassLoader.loadClass(AntClassLoader.java:1090)
        ... 84 more
```

**Comment by snicoll** [comment]
@lasalazarr I can't reproduce the issue and looking at the surface area I doubt something has changed since RC1. My best guess is that a jar got corrupted when you upgraded. Can you please run `mvn dependency:purge-local-repository` (or delete your local m2 repo if that doesn't help)?

**Comment by lasalazarr** [comment]
The problem even cleaning Gradle Cache continue. 

`rm -rf .gradle/caches/`

And also run: 

`BUILD FAILED in 13s
`

The environment it is:

`java version "9"
Java(TM) SE Runtime Environment (build 9+181)
Java HotSpot(TM) 64-Bit Server VM (build 9+181, mixed mode)
------------------------------------------------------------
Gradle 4.4.1
------------------------------------------------------------

Build time:   2017-12-20 15:45:23 UTC
Revision:     10ed9dc355dc39f6307cc98fbd8cea314bdd381c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          9 (Oracle Corporation 9+181)
OS:           Mac OS X 10.13.3 x86_64

`
I just push a simple sample project for you can reproduce the problem:

[Sample Project to Reproduce the problem](https://github.com/lasalazarr/springboot2-ws-soap-consume)

**Comment by spencergibb** [comment]
https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-Java-9#jaxb

**Comment by snicoll** [comment]
@lasalazarr thank you for the sample. The difference between RC1 and `2.0.0.RELEASE` is that [we provide dependency management for jaxb now](https://github.com/spring-projects/spring-boot/issues/11205) (bumped to 2.3.0).

That dependency hardcoded on 2.1.7 is no longer legit and we still need to update the samples to Spring Boot 2. I've tried to bump that dependency locally but I got a different error. I am not sure yet if that's a specific issue in Spring Boot.

**Comment by lasalazarr** [comment]
Thanks; I will test with just Spring WS to see if I figure out the problem; but even with 	jaxb "com.sun.xml.bind:jaxb-xjc:2.3.0"
 + Spring Boot 2.0.0.RELEASE it is not working.

**Comment by wilkinsona** [comment]
For reasons that I don't understand, in 2.3.0 the pom for `com.sun.xml.bind:jaxb-xjc` doesn't declare its dependencies so you have to assemble all of them manually. On Java 9, you need the following dependencies:

```
	jaxb "com.sun.xml.bind:jaxb-core:2.3.0"
	jaxb "com.sun.xml.bind:jaxb-impl:2.3.0"
	jaxb "com.sun.xml.bind:jaxb-xjc:2.3.0"
	jaxb "javax.xml.bind:jaxb-api"
	jaxb "javax.activation:activation:1.1"
```

This resolves the class path related problems, but `xjc` still fails and without any useful diagnostics:

```
./gradlew clean build --stacktrace

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/awilkinson/dev/temp/springboot2-ws-soap-consume/build.gradle' line: 49

* What went wrong:
Execution failed for task ':genJaxb'.
> xjc failed

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':genJaxb'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: : xjc failed
        at com.sun.tools.xjc.XJCBase.execute(XJCBase.java:731)
        at com.sun.tools.xjc.XJC2Task.execute(XJC2Task.java:55)
        at com.sun.istack.tools.ProtectedTask.execute(ProtectedTask.java:103)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:78)
        at org.gradle.api.internal.project.ant.BasicAntBuilder.doInvokeMethod(BasicAntBuilder.java:103)
        at org.gradle.internal.metaobject.BeanDynamicObject$GroovyObjectAdapter.invokeOpaqueMethod(BeanDynamicObject.java:579)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.invokeMethod(BeanDynamicObject.java:506)
        at org.gradle.internal.metaobject.BeanDynamicObject.tryInvokeMethod(BeanDynamicObject.java:191)
        at org.gradle.internal.metaobject.ConfigureDelegate.invokeMethod(ConfigureDelegate.java:57)
        at build_70cxwhfrghjbqbj9u427z31nk$_run_closure3$_closure11$_closure12.doCall(/Users/awilkinson/dev/temp/springboot2-ws-soap-consume/build.gradle:49)
        at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:71)
        at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:160)
        at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:106)
        at org.gradle.api.internal.project.DefaultProject.ant(DefaultProject.java:1107)
        at org.gradle.api.Project$ant.call(Unknown Source)
        at build_70cxwhfrghjbqbj9u427z31nk$_run_closure3$_closure11.doCall(/Users/awilkinson/dev/temp/springboot2-ws-soap-consume/build.gradle:43)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:726)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:699)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 30 more


* Get more help at https://help.gradle.org

BUILD FAILED in 1s
2 actionable tasks: 2 executed
```

This appears to be the problem described in [this issue](https://github.com/javaee/jaxb-v2/issues/1144). That issue suggests setting `fork` to `true`. Doing so doesn't fix the problem, but with debug output enabled it then becomes apparent that the wrong command line is being built:

```
/Library/Java/JavaVirtualMachines/jdk-9.jdk/Contents/Home/bin/java -classpath /Users/awilkinson/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-core/2.3.0/d044c784e41d026778693fb44a8026c1fd9a7506/jaxb-core-2.3.0.jar:/Users/awilkinson/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-impl/2.3.0/3a3c6a62719d967175b76b63925f1fb495f11437/jaxb-impl-2.3.0.jar:/Users/awilkinson/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-xjc/2.3.0/caabab034b247d72b91ed374a3191a66511dd7e1/jaxb-xjc-2.3.0.jar:/Users/awilkinson/.gradle/caches/modules-2/files-2.1/javax.xml.bind/jaxb-api/2.3.0/99f802e0cb3e953ba3d6e698795c4aeb98d37c48/jaxb-api-2.3.0.jar:/Users/awilkinson/.gradle/caches/modules-2/files-2.1/javax.activation/activation/1.1/e6cb541461c2834bdea3eb920f1884d1eb508b50/activation-1.1.jar com.sun.tools.xjc.XJCFacade -d /Users/awilkinson/dev/temp/springboot2-ws-soap-consume/build/generated-sources/jaxb -p hello.wsdl -wsdl /Users/awilkinson/dev/temp/springboot2-ws-soap-consume/http:/www.webservicex.com/stockquote.asmx?WSDL
```

Note the `/Users/awilkinson/dev/temp/springboot2-ws-soap-consume/http:/www.webservicex.com/stockquote.asmx?WSDL` at the end of the command. It should just be `http:/www.webservicex.com/stockquote.asmx?WSDL`

In summary the XJC 2.3 Ant Task appears to not support remote schema. Until this is fixed, you can either use a local schema file with fork enabled or you could downgrade to the most recent version that works. That appears to be 2.2.11 which can be configured with the following dependencies:

```
	jaxb "com.sun.xml.bind:jaxb-core:2.2.11"
	jaxb "com.sun.xml.bind:jaxb-impl:2.2.11"
	jaxb "com.sun.xml.bind:jaxb-xjc:2.2.11"
	jaxb "javax.xml.bind:jaxb-api:2.2.11"
	jaxb "javax.activation:activation:1.1"
```

**Comment by snicoll** [comment]
@lasalazarr we'll have a look to the getting started guide you've referenced as well as we are in the process of upgrading them all to Spring Boot 2.

**Comment by wilkinsona** [comment]
The Guides currently target Java 8 so, rather than complicating things for everyone by dealing with the problems with XJC and Java 9, I think we should leave things as-is in the hope that the problems with XJC are addressed soon.

**Comment by wilkinsona** [comment]
I've just learned about `org.glassfish.jaxb:jaxb-xjc-jdk9`. It has the same problem with the remote schema, but it does require fewer dependencies to use it:

```
	jaxb "org.glassfish.jaxb:jaxb-xjc-jdk9:2.3.0"
	jaxb "javax.activation:activation:1.1"
```
